### DISCUSSION TITLE
unable to convert NEF? TIF? file

### DISCUSSION THREAD

**Discussion by ryadical** [ImageMagick/ImageMagick#2128] [Help]
I have a legacy medical system whose vendor is out of business. The system allowed users to scan documents and index them and the data was stored in a Postgre DB. We are trying to export the images from the DB and the majority of them can not be opened/viewed when written from the byte array in postgre to a binary file, but some can. The application scanned multipage documents into individual images and allowed annotations on the page. (stored in a separate file) There is no field on these that specifies extension but I had assumed TIF based on the first two characters of the file starting with "II".

I am looking for an example image without PHI on it that I could provide but I don't have one as of yet. I can't find any application that will view or convert it. All of the images can be viewed from within the client application. I tried naming the file TIF and NEF and I get different errors on Magick converting it. I downloaded some sample NEF files from the internet and had no issue converting them. (compression field from exif looked exactly the same)

`c:\Program Files\ImageMagick-7.0.10-Q8\magick.exe h:\soapware.tif h:\soapware.pdf
magick.exe: compression not supported `h:\soapware.tif' @ error/tiff.c/ReadTIFFImage/1615.

c:\Program Files\ImageMagick-7.0.10-Q8\magick.exe h:\soapware.nef h:\soapware.pdf
magick.exe: Unsupported file format or not RAW file `h:\soapware.nef' @ error/dng.c/ReadDNGImage/420.`

This is 100% of the exif data for one of the files:

File Name                       : soapware.tif
Directory                       : h:/
File Size                       : 64 kB
File Modification Date/Time     : 2020:06:09 14:10:51-04:00
File Access Date/Time           : 2020:06:09 15:37:05-04:00
File Creation Date/Time         : 2020:06:09 15:37:05-04:00
File Permissions                : rw-rw-rw-
File Type                       : NEF
File Type Extension             : nef
MIME Type                       : image/x-nikon-nef
Exif Byte Order                 : Little-endian (Intel, II)
Subfile Type                    : Single page of multi-page image
Image Width                     : 2544
Image Height                    : 3300
Bits Per Sample                 : 1
Compression                     : Nikon NEF Compressed
Photometric Interpretation      : BlackIsZero
Strip Offsets                   : 234
Orientation                     : Horizontal (normal)
Samples Per Pixel               : 1
Rows Per Strip                  : 3300
Strip Byte Counts               : 46478
X Resolution                    : 300
Y Resolution                    : 300
Planar Configuration            : Chunky
Resolution Unit                 : inches
Page Number                     : 0 0
Image Size                      : 2544x3300
Megapixels                      : 8.4

**Comment by snibgo** [comment]
Your file claims to be 2544x3300 pixels, but only 1 bit per pixel. It is 8.4 MB. And it is compressed, it claims.

Those numbers don't add up. 2544*3300 = 8395200. So it has 8.4 M pixels. So that suggests the file is 1 byte/pixel, not 1 bit/pixel, and that it isn't compressed.

Perhaps the metadata is deliberately wrong to prevent other software from reading it. 

You could try changing metadata with exiftool, and see if IM can then read it.

**Reply by ryadical** [reply]
The file is actually 64k according to windows. In reality it is an 8.5x11 black/white scanned document. 

I am new all of this, any suggestions on what to change?

**Comment by snibgo** [comment]
Sorry, I misread the metadata. I thought it said "8.4 megabytes". Ignore my previous post.

I have never seen a 1-bit image with NEF compression, so that might be wrong. But I don't know.
